core/src/main/java/org/elasticsearch/bootstrap/Bootstrap.java

```diff
@@ -297,6 +298,13 @@ static void init(String[] args) throws Throwable {
         }
     }
 
+    @SuppressForbidden(reason = "Sets system properties passed as CLI parameters")
+    private static void elasticsearchSettings(Map<String, String> esSettings) {
+        for (Map.Entry<String, String> esSetting : esSettings.entrySet()) {
+            System.setProperty(esSetting.getKey(), esSetting.getValue());
```

Comment:
@rjernst We do to it today, from the [new bootstrap CLI parser](https://github.com/elastic/elasticsearch/blob/e4bed0c97e7e0acb293624f9601027f8be38c874/core/src/main/java/org/elasticsearch/bootstrap/BootstrapCliParser.java#L87):

```java
        for (String property : propertyOption.values(options)) {
            String[] keyValue = property.split("=", 2);
            if (keyValue.length != 2) {
                throw new UserError(ExitCodes.USAGE, "Malformed elasticsearch setting, must be of the form key=value");
            }
            String key = keyValue[0];
            if (key.startsWith("es.") == false) {
                key = "es." + key;
            }
            System.setProperty(key, keyValue[1]);
        }
```

I *very* much want to get rid of it, but I consider it outside of the scope of this pull request. My [comment](https://github.com/elastic/elasticsearch/pull/16791#issuecomment-188171755) from the original pull request which this pull request supersedes:

> This pull request is complete, but the work here is not. In particular, the eventual end game is to remove the use of setting system properties to get command line arguments from the CLI parser down to the bootstrap process. The pieces are in place for this to be a simple extension of this pull request, but there is a lot of time-consuming tedious work to be done to change all the places where we are setting system properties (tests, docs, etc.). This can be done in a follow-up pull request.